### PR TITLE
config/: Add support for filtering out detected port scans via scanlogd.

### PR DESCRIPTION
**Pull request by sunweaver**
Before submitting your PR, please review the following checklist:

- [x] **CHOOSE CORRECT BRANCH**: if filing a bugfix/enhancement
      against certain release version, choose `0.9`, `0.10` or `0.11` branch,
      for dev-edition use `master` branch
- [-] **CONSIDER adding a unit test** if your PR resolves an issue
- [-] **LIST ISSUES** this PR resolves
- [ ] **MAKE SURE** this PR doesn't break existing tests
- [x] **KEEP PR small** so it could be easily reviewed.
- [x] **AVOID** making unnecessary stylistic changes in unrelated code
- [-] **ACCOMPANY** each new `failregex` for filter `X` with sample log lines
      within `fail2ban/tests/files/logs/X` file

### CODE REVIEW

**Review by szepeviktor — APPROVED** [review]

File: config/filter.d/scanlogd.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Fail2Ban filter for port scans detected by scanlogd
+
+[Definition]
+
+failregex = scanlogd:\ <HOST>\ to\ .*\ ports\ .*
```
Comment (sebres) [inline config/filter.d/scanlogd.conf:5]: this RE is vulnerable due to catch-all (`.*`) and missing anchors; we don't accept filters with unjustified catch-all's anymore

**Review by sebres — CHANGES_REQUESTED** [review]
please fix the RE vulnerability as well as provide a test cases for lines matching this RE in [fail2ban/tests/files/logs/scanlogd](https://github.com/sunweaver/fail2ban/new/pr/scanlogd-filter/fail2ban/tests/files/logs)

File: config/filter.d/scanlogd.conf (hunk 1 above)
Comment (sunweaver) [reply config/filter.d/scanlogd.conf]: @sebres: ack, PR amended. Please revisit.

**Comment by szepeviktor** [comment]
Definitely getting better.

File: config/filter.d/scanlogd.conf (hunk 1 above)
Comment (sebres) [reply config/filter.d/scanlogd.conf]: There are still no tests in fail2ban/tests/files/logs/scanlogd, so test suite fails in CI.
Either add the tests or provide a log excerpt with messages the filter match (then I'll add it to the PR by myself).

**Comment by sunweaver** [comment]
Thanks for providing the tests. Here are some log lines take from a live server (replace <server> with the hostname, <badhost-IP> and <my-IP> with IPv4/IPv6 addresses

```
/var/log/syslog.3.gz:Mar  5 21:44:43 <server> scanlogd: <badhost-IP> to <my-IP> ports 80, 81, 83, 88, 99, 443, 1080, 3128, ..., f????uxy, TOS 00, TTL 49 @20:44:43
/var/log/syslog.4.gz:Mar  4 10:04:25 <server> scanlogd: <badhost-IP> to <my-IP> ports 497, 515, 544, 543, 464, 513, ..., fSrpauxy, TOS 00 @09:04:25
/var/log/syslog.4.gz:Mar  4 18:34:00 <server> scanlogd: <badhost-IP> to <my-IP> ports 593, 548, 636, 646, 625, 631, ..., fSrpauxy, TOS 00, TTL 239 @17:34:00
/var/log/syslog.6.gz:Mar  2 23:38:37 <server> scanlogd: <badhost-IP> to <my-IP> ports 22, 26, 37, 80, 25, 79, ..., fSrpauxy, TOS 00 @22:38:37
```

In fact, I just tested the recent release of scanlogd and found that it does not report IPv6 port scans. Will file an upstream bug report for this.

**Review by sebres — APPROVED** [review]

**Pull request merged**